autonomous/forward.py

```diff
@@ -0,0 +1,19 @@
+from magicbot import AutonomousStateMachine, state
+from components.drivetrain import Drivetrain
+
+
+class Forward(AutonomousStateMachine):
+    MODE_NAME = "Forward"
+    DEFAULT = True
+
+    drivetrain = Drivetrain
+
+    @state(first=True)
+    def forward(self):
+        self.drivetrain.forward(feet=25)
+        self.next_state('wait_till_completion')
+
+    @state
+    def wait_till_completion(self):
```

Comment:
I'm not sure I like this. Could we do something like this instead?

```python
@state(first=True)
def forward(self):
    done = self.drivetrain.forward(feet=25)
     if done:
          self.next_state('whatever')
```
This means that `drivetrain.forward` would be called constantly, and would return a boolean, whether it is done. I think we should `update` the `ProfileExecutor` when `forward` and `rotate` are called, rather than in `execute`

Reply:
Hmm. So what should happen if `drivetrain.forward(feet=20)` had been called, and then `drivetrain.forward(feet=19)` was called?

Reply:
It wouldn't override.

Reply:
So after the first call to `drivetrain.forward(feet=_)`, the parameters are just ignored?

Reply:
Actually, we could keep the goal distance (or angle) on self, and it would override if it is different than the current goal distance

Reply:
Ok. I can live with that. Since there are multiple distance parameters to `drivetrain.forward`, does wrapping them in a tuple and storing them that way seem good? Alternatively, we *could* just ignore the arguments and force a call to `drivetrain.cancel_motion_profile()`.

Reply:
Just combine them all to metric and save the metric value.

Reply:
Rather then using `drivetrain.cancel_motion_profile` to switch profiles?

Reply:
I was thinking, we really shouldn't ever have to switch profiles, so storing the arguments in expectation of that seems a bit weird. Also it screws with expected behavior, as calling `drivetrain.forward` will sometimes override, sometimes not. As another option (I'm really just throwing them out so we consider everything), there could be some `override` parameter specifying whether you want to override existing profiles.

Reply:
Yeah, I guess cancelling might be a better path here. Maybe (in this PR or later on) error or warn if the parameters are changed without cancelling?

Reply:
Ok, sounds good. I think warning is a good idea.